File: AbstractFactoryDesignPattern/NotificationService_Example/Driver.java
package AbstractFactoryDesignPattern.NotificationService_Example;

import AbstractFactoryDesignPattern.NotificationService_Example.Factory.Factory;
import AbstractFactoryDesignPattern.NotificationService_Example.Factory.NotificationFactory;
import AbstractFactoryDesignPattern.NotificationService_Example.Message.Message;
import AbstractFactoryDesignPattern.NotificationService_Example.Sender.Sender;

import java.util.Scanner;

public class Driver {
	public static void main(String[] args) {
		Scanner scannerObj = new Scanner(System.in);

		System.out.println("Welcome to notification Service. Please Select Mode of Notification: ");
		System.out.println("1. email\n2. sms");
		String notificationMode = scannerObj.nextLine();

		Factory factory = NotificationFactory.createFactory(notificationMode);
		Message message = factory.createMessage();

		System.out.println("Enter the Following Details: ");
		System.out.println("Note: SMS mode SMS Body has a Character restriction of 160 characters.");
		System.out.println("Recipient Info (phoneNumber or email): ");
		String recipient = scannerObj.nextLine();

		System.out.println("Message Body: ");
		String body = scannerObj.nextLine();

		message.setContent(recipient, body);

		Sender sender = factory.createSenderService();
		sender.sendMessage(message);
		System.out.println("Message sent successfully.");
	}
}

File: AbstractFactoryDesignPattern/NotificationService_Example_Python/Driver.py
import sys

from AbstractFactoryDesignPattern.NotificationService_Example_Python.Factory.AbstractNotificationFactory import \
	AbstractNotificationFactory
from AbstractFactoryDesignPattern.NotificationService_Example_Python.Factory.InterfaceFactory import InterfaceFactory
from AbstractFactoryDesignPattern.NotificationService_Example_Python.Message.NotificationMessage import \
	NotificationMessage
from AbstractFactoryDesignPattern.NotificationService_Example_Python.SenderService.NotificationService import \
	NotificationService


class Driver:

	@staticmethod
	def main() -> None:
		userInput: str = ""
		while userInput != "exit":
			print("""Please Select the Mode of Notification:
			1. email
			2. sms""")

			inputMode : str = sys.stdin.readline()
			userModeInput : str = inputMode if len(inputMode) > 1 else "not given"
			notifyFactory : InterfaceFactory = AbstractNotificationFactory.getNotificationFactory(userModeInput.lower().strip())

			message : NotificationMessage = notifyFactory.createMessage()
			messageSendingService : NotificationService = notifyFactory.createSenderService()

			print("Please provide the following details:")
			print("Note: the SMS message can only be 160 Character Long.")
			print("Recipient Phone Number or Email Address:")
			inputRecipient: str = sys.stdin.readline()
			print("Message for Recipient:")
			inputMessageBody: str = sys.stdin.readline()
			message.setContent(inputRecipient.lower().strip(), inputMessageBody.strip())
			messageSendingService.sendMessage(message)

			print("If you want to exit please type exit. Otherwise press enter.")
			userInput: str = sys.stdin.readline()
			userInput = userInput.lower().strip()

		sys.exit(0)

if __name__ == "__main__":
	Driver.main()